api/src/pcapi/core/educational/api/dms.py

```diff
@@ -0,0 +1,165 @@
+from datetime import datetime
+from datetime import timedelta
+import logging
+import typing
+
+from pcapi import settings
+from pcapi.connectors.dms import api as dms_api
+from pcapi.connectors.dms import models as dms_models
+from pcapi.core.educational import models as educational_models
+from pcapi.core.offerers import models as offerers_models
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+
+def update_dms_status(procedure_id: int = 0) -> None:
+    """import dms applications for eac status"""
+    if procedure_id:
+        procedures = [procedure_id]
+    else:
+        procedures = [
+            settings.DMS_EAC_PROCEDURE_INDEPENDANTS_CANDIDATE_ID,
+            settings.DMS_EAC_PROCEDURE_STRUCTURE_CANDIDATE_ID,
+            settings.DMS_EAC_PROCEDURE_MENJS_CANDIDATE_ID,
+        ]
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ça peut être un peu plus simple à lire si on n'a pas cette logique, et dans la commande on fait 

```python
for procedure_id in [
            settings.DMS_EAC_PROCEDURE_INDEPENDANTS_CANDIDATE_ID,
            settings.DMS_EAC_PROCEDURE_STRUCTURE_CANDIDATE_ID,
            settings.DMS_EAC_PROCEDURE_MENJS_CANDIDATE_ID,
        ]:
    update_dms_status(procedure_id)

Reply:
ça oblige a faire de la copie de code, je suis pas hyper fan

Reply:
ça ne duplique pas de code a priori
ça ne fais que déplacer la rsponsabilité de choisir la procédure à la commande au lieu d'avoir ce switch dans la fonction

tu aurais: 

```python 

# api/src/pcapi/core/educational/api/dms.py

def import_dms_application(procedure_number) -> None:
        previous_import = _get_previous_import(procedure)
        if previous_import and previous_import.isProcessing:
            logger.info(
                "[DMS] Procedure %s is already being processed.",
                procedure,
                extra={
                    "procedure_number": procedure,
                },
            )
            continue
        since = previous_import.latestImportDatetime if previous_import else None

        update_dms_applications_for_procedure(
            procedure_number=procedure,
            since=since,
        )


# in commands.py

@blueprint.cli.command("import_eac_dms_application")
@log_cron_with_transaction
def import_eac_dms_application() -> None:
    procedures = [
        settings.DMS_EAC_PROCEDURE_INDEPENDANTS_CANDIDATE_ID,
        settings.DMS_EAC_PROCEDURE_STRUCTURE_CANDIDATE_ID,
        settings.DMS_EAC_PROCEDURE_MENJS_CANDIDATE_ID,
    ]
    for procedure in procedures:
        import_dms_application(procedure)
```

Reply:
Edit: en écrivant mon commentaire ci-dessus, je passe en high logaf sur ce point, je pense vraiment que c'est mieux

-> Single responsibility de la fonction `import_dms_application`

Reply:
Je suis d'accord avec Corentin, je trouve préférable que la fonction ne se charge que d'une seule procédure.

Reply:
ha oui en effet je n'avais pas pensé a le déplacer dans la commande. Je prend cette correction :)